### PR TITLE
StableDiffusionInpaintingPipeline - resize image w.r.t height and width

### PR DESCRIPTION
**Pull request by rupertmenneer**
See issue https://github.com/huggingface/diffusers/issues/3103

The problem is that images of arbitrary dimensions can be passed into the `StableDiffusionInpaintingPipeline` and this can result in a tensor mismatch error. Parameters already exist for desired height/width so this PR simply resizes the incoming image and mask to the passed parameters which default to 512 for this pipeline.

Additionally, I moved a simple 'is None' check of the images/masks into the `prepare_mask_and_masked_image` function as this function already handles the incoming images and checks for typing/content errors. This simply tidies up the main body of the `__call__` function which is quite bloated.

I have tested this new change on this pipeline and it works as expected. Test code:

```
# different aspect ratios
from diffusers import StableDiffusionInpaintPipeline
height, width = 480, 480
pipe = StableDiffusionInpaintPipeline.from_pretrained(
    "stabilityai/stable-diffusion-2-inpainting",
    torch_dtype=torch.float16,
)
pipe.to("cuda")
prompt = "a mountain landscape"
output_image = pipe(prompt=prompt,
            image=landscape_image,
            mask_image=landscape_mask,
            height=height,
            width=width,
            num_inference_steps=25).images[0]

print(f"Original image size {landscape_image.size}")
print(f"Output image size {output_image.size}")

output_image
```

Produces:

Screenshot 2023-05-03 at 11 45 27

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py [hunk 1]
```diff
@@ -787,12 +796,6 @@ def __call__(
             negative_prompt_embeds,
         )
 
-        if image is None:
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py:790]: I think we should leave those, inpainting needs both image and mask_image

**Review by patrickvonplaten — APPROVED** [review]
Very cool! IMO different behavior between PIL (allows resizing) and PT doesn't allow it is fine here. 

@sayakpaul and @yiyixuxu can you also check here? :-)

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py [hunk 2]
```diff
@@ -109,8 +116,9 @@ def prepare_mask_and_masked_image(image, mask):
         # preprocess image
         if isinstance(image, (PIL.Image.Image, np.ndarray)):
             image = [image]
-
         if isinstance(image, list) and isinstance(image[0], PIL.Image.Image):
+            # resize all images w.r.t passed height an width
+            image = [i.resize((width, height), resample=PIL.Image.LANCZOS) for i in image]
```
Comment (sayakpaul) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py:121]: If it's ndarray then this resizing method won't work. But fine for me to leave that case in this PR.

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py [hunk 3]
```diff
@@ -109,8 +116,9 @@ def prepare_mask_and_masked_image(image, mask):
         # preprocess image
         if isinstance(image, (PIL.Image.Image, np.ndarray)):
             image = [image]
-
         if isinstance(image, list) and isinstance(image[0], PIL.Image.Image):
+            # resize all images w.r.t passed height an width
+            image = [i.resize((width, height), resample=PIL.Image.LANCZOS) for i in image] 
             image = [np.array(i.convert("RGB"))[None, :] for i in image]
             image = np.concatenate(image, axis=0)
```
Comment (sayakpaul) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py:123]: A natural extension to the above conditional branches, IMO, is what happens when:

```python
elif isinstance(image, list) isinstance(image[0], np.ndarray):
```

We could convert to PIL from the ndarrays and then unify the logic here.

File: tests/pipelines/stable_diffusion/test_stable_diffusion_inpaint.py [hunk 4]
```diff
@@ -300,6 +300,25 @@ def test_inpaint_compile(self):
...
+            )
+            pipe.scheduler = LMSDiscreteScheduler.from_config(pipe.scheduler.config)
+            pipe.to(torch_device)
+            pipe.set_progress_bar_config(disable=None)
+            pipe.enable_attention_slicing()
+
+            inputs = self.get_inputs(torch_device)
+            # change input image to a random size (one that would cause a tensor mismatch error)
+            inputs['image'] = inputs['image'].resize((127,127))
+            inputs['mask_image'] = inputs['mask_image'].resize((127,127))
```
Comment (sayakpaul) [inline tests/pipelines/stable_diffusion/test_stable_diffusion_inpaint.py:315]: Why 127?

**Review by sayakpaul — APPROVED** [review]
LGTM! Thank you! 

I think we need to make sure the code style checks pass on the CI before we can merge this. 

> added a new unit test which passes in an incorrectly sized input image into the pipeline (which used to throw a tensor mismatch error), this now returns an image of the correct size w.r.t passed height/width

Very cool!

> Final point, just want to mention that these two unit tests are failing on my setup on the main branch, just wanted to check that others are getting the same? if so I will setup a separate issue

What is your current setup? Are you using Torch 2.0?

**Comment by patrickvonplaten** [comment]
Fixing `make style` on main

**Pull request merged**